projects/templatetags/projects_extras.py

```diff
@@ -0,0 +1,80 @@
+from django import template
+from projects.models import *
+
+register = template.Library()
+
+
+@register.filter
+def get_bundles(project):
+    return BundleDetail.objects.filter(project__id=project.id).exclude(status='ARCHIVED')
+
+
+@register.filter
+def get_builds(bundle):
+    return BuildDetail.objects.filter(bundle__id=bundle.id).exclude(status='ARCHIVED')
+
+
+@register.filter
+def get_suite(bundle):
+    return SuiteDetail.objects.filter(bundle__id=bundle.id).exclude(status='ARCHIVED')[0]
+
+
+@register.filter
+def get_tests(suite):
+    return TestScenarioDetail.objects.filter(suite__id=suite.id).exclude(status='ARCHIVED')
+
+
+@register.filter
+def get_build_tests(build):
+    return BuildTestDetail.objects.filter(build__id=build.id).exclude(status='ARCHIVED')
+
+
+@register.filter
+def check_failure(build):
+    for test in get_build_tests(build):
+        if test.result == 'FAILED':
+            return 'FAILED'
+        elif test.result == 'NONE':
+            return 'INCOMPLETE'
+    return 'PASSED'
+
+
+@register.filter
+def get_issues(build=None, build_test=None):
+    tests = []
+    if build is not None:
+        tests = get_build_tests(build)
+    if build_test is not None:
+        tests.append(build_test)
+    return IssueDetail.objects.filter(build_test__id__in=tests.id).exclude(status='ARCHIVED')
+
+
+@register.filter
+def generate_breadcrumbs(instance, f=True):
```

Comment:
maybe give the local one something more descriptive. I assume `final` means it's the last breadcrumb? why not `final_breadcrumb`?